public/chat_header_button.tsx

```diff
@@ -119,12 +122,36 @@ export const HeaderChatButton = (props: HeaderChatButtonProps) => {
     }
   };
 
+  useEffect(() => {
+    if (!flyoutLoaded && flyoutVisible) {
+      const mountPoint = flyoutMountPoint.current;
+      if (mountPoint) {
+        core.overlays.sidecar().open(mountPoint, {
+          className: 'chatbot-sidecar',
+          config: {
+            dockedMode: SIDECAR_DOCKED_MODE.RIGHT,
+            paddingSize: 460,
+          },
+        });
+        flyoutLoaded = true;
+      }
+    } else if (flyoutLoaded && flyoutVisible) {
+      core.overlays.sidecar().show();
+    } else if (flyoutLoaded && !flyoutVisible) {
+      core.overlays.sidecar().hide();
+    }
+  }, [flyoutVisible, flyoutLoaded, flyoutMountPoint]);
+
   const onKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
     if (e.key === 'Escape') {
       inputRef.current?.blur();
     }
   };
 
+  const setMountPoint = useCallback((mountPoint) => {
+    flyoutMountPoint.current = mountPoint;
```

Comment:
Shall we check if flyoutVisible here. We may need to call `core.overlays.sidecar().open` here if, flyoutVisible is `true` and previous `flyoutMountPoint.current` is null.

Reply:
Can you elaborate more? I think we don't need to do more judgement in setMountPoint. Some necessary judgements are done when calling `sidecar().open()`

Reply:
The `sidebar().open()` in `useEffect` will be called if `flyoutVisible` equal true and `mountPoint` is truly. It won't be called if the mountPoint is null. Shall we add related logic for this case? The logic should be move to `setMountPoint` if we need to handle this.